src/IndexedDB/Browser.IndexedDB.fs

```diff
@@ -0,0 +1,143 @@
+﻿namespace rec Browser.Types
+
+open System
+open Fable.Core
+
+
+[<Erase>]
+type IDBRequestSource =
+    | Index of IDBIndex
+    | ObjectStore of IDBObjectStore
+    | Cursor of IDBCursor
```

Comment:
This should work. When using `Erase` attribute, Fable will replace pattern match with type testing. In the case of interfaces, if they're decorated with `Global` Fable will use `instanceof IDBIndex`.